docker/docker-compose.yml

```diff
@@ -1,41 +1,49 @@
-version: '3.2'
-services:
-
-  jenkinsserver:
-    build: ./jenkinsserver
-    ports:
-      - "80:8080"
-#    restart: always
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
-      - type: volume
-        source: docker_jenkinsdata
-        target: /var/jenkins_home
-        volume:
-          nocopy: true
-    environment:
-      JAVA_OPTS: -Duser.timezone=Europe/Stockholm
+version: '2'
 
-  gitserver:
-    build: ./gitserver
-    restart: always
-    container_name: git-server
-    ports:
-      - "2222:22"
-    volumes:
-      - type: volume
-        source: docker_gitdata
-        target: /git-server/repos
-        volume:
-          nocopy: true
-      - type: volume
-        source: docker_keys
-        target: /git-server/keys
-        volume:
-          nocopy: true
+services:
+    jenkins-master:
+        build: jenkins-master
+        volumes:
+            - jenkins-home:/var/jenkins_home
+            - jenkins-log:/var/log/jenkins
+            - /var/run/docker.sock:/var/run/docker.sock
+            - git-server-keys:/git-server/keys
+            - ~/Workspace:/projects:ro
+        ports:
+            - "80:8080"
+    git-server:
+        build: git-server
+        depends_on:
+            - jenkins-master
+        ports:
+            - "2222:22"
+        volumes:
+            - git-server-keys:/git-server/keys
+            - git-server-repos:/git-server/repos
+    jenkins-agent01:
```

Comment:
Behöver vi verkligen slavar? Eller tycker du bara det är bra?

Reply:
Nä, ärligt talat följde de med bara för att jag redan lagt in dem i min egen uppsättning. Men tror du att det skadar att ha med dem? Så småningom lär väl ämnet dyka upp?